packages/lib-classifier/src/store/DrawingStore.js

```diff
@@ -1,56 +1,156 @@
-import { addDisposer, getRoot, onAction, types } from 'mobx-state-tree'
+import cuid from 'cuid'
+import { addDisposer, getRoot, getType, onAction, types } from 'mobx-state-tree'
 import { autorun } from 'mobx'
 import { Subject } from 'rxjs'
 
+import { Line, Point } from './markings'
+
+const markModels = {
+  line: Line,
+  point: Point
+}
+
+function getMarkModel (toolType) {
+  return markModels[toolType] || null
+}
+
 const DrawingStore = types
   .model('DrawingStore', {
-    activeDrawingTool: types.optional(types.number, 0)
+    activeDrawingToolIndex: types.optional(types.number, 0),
+    activeMark: types.safeReference(types.union({
+      dispatcher: getType
+    }, Line, Point)),
+    marks: types.map(types.union({
```

Comment:
This is a repeat of the types declaration from the previous line. They could be rolled up into a single piece of code.

Reply:
Resolved in #1266.